crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -113,7 +112,7 @@ pub(super) struct SlidingSyncInner {
     room_subscriptions: StdRwLock<BTreeMap<OwnedRoomId, v4::RoomSubscription>>,
 
     /// Rooms to unsubscribe, see [`Self::room_subscriptions`].
-    room_unsubscriptions: StdRwLock<Vec<OwnedRoomId>>,
+    room_unsubscriptions: StdRwLock<HashSet<OwnedRoomId>>,
```

Comment:
Let's use `BTreeSet` for consistency (will also be easier for testing since the list will have a deterministic order).